src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -768,14 +770,35 @@ pub enum StaleItem {
 impl LocalFingerprint {
     /// Read the environment variable of the given env `key`, and creates a new
     /// [`LocalFingerprint::RerunIfEnvChanged`] for it.
-    ///
-    // TODO: This is allowed at this moment. Should figure out if it makes
-    // sense if permitting to read env from the config system.
```

Comment:
Instead of removing this, can we have a comment explaining why the `allow` can't be removed? It's not obvious to me that there is some fundamental problem that is preventing this.

I presume the problem is that the closure cannot access the GlobalContext, and thus is unable to access the env snapshot, and thus must access the raw environment directly? It seems like there are still options we can consider, such as wrapping the environment in Arc or something like that?

Reply:
> such as wrapping the environment in Arc

Agree `Arc` is the way to resolve this.